src/backend/distributed/sql/citus--9.2-1--9.2-2.sql

```diff
@@ -1,2 +1,5 @@
 -- reserve UINT32_MAX (4294967295) for a special node
 ALTER SEQUENCE pg_catalog.pg_dist_node_nodeid_seq MAXVALUE 4294967294;
+
+GRANT EXECUTE ON FUNCTION pg_catalog.worker_create_schema(bigint) TO PUBLIC;
+GRANT EXECUTE ON FUNCTION pg_catalog.worker_repartition_cleanup(bigint) TO PUBLIC;
```

Comment:
This seems a bit hazardous. The schemas all sit in the same namespace, so unprivileged user A can drop/create a job schema for a query being run by privileged user B.

One solution could be to add the user IDs to the job schema name, similar to how we add the user ID to the intermediate result directory and task files. That way, each user has its own namespace and data can only leak between transactions of the same user.

Reply:
An alternative could be to pass the user ID to worker_create_schema and grant CREATE permission to the user, or separately send a GRANT CREATE as superuser.

Reply:
Now passing user name to this function and executing as citus extension owner.